src/Cart.java

```diff
@@ -10,9 +10,27 @@ public Cart() {
         this.productsInCart = new ArrayList<>();
     }
 
+    //metodo geralizzato che verifica prima se si può finalizzare la vendita
+    // e poi procede svuotando il carrello e rimuovendo i device venduti dal magazzino;
+
+    public static void finalizeSellingProcess(boolean paymentCheck, List<Product> productsInCart, List<Product> productsInStock) throws IllegalAccessException {
+        if (paymentCheck) {
+
+            productsInStock.removeAll(productsInCart);
+            productsInCart.clear();
+        } else {
+            throw new IllegalAccessException("you must first complete a form of payment");
```

Comment:
In questo caso, se mai dovesse partire l'eccezione non gestita con try and catch manderebbe in shutdown il programma. Ciò non dovrebbe mai accadere.

Reply:
e cosa ti metto ?una stampa e basta o vuoi che implemento un try catch nel metodo al posto dell'eccezione ?

Reply:
Dipende, in realtà non so neanche se sia necessario gestire questa eccezione, poichè pensaci: anche se il carrello è vuoto e questo metodo passa, che male succede? svuoti un carrello già vuoto ed elimini dal warehouse nessun prodotto non essendoci prodotti nel carrello. Eventuali controlli di questo tipo li inserirei postumi nel momento in cui incappiamo in problemi. Ci sono molti controlli da fare, che ancora non sono stati inseriti, quindi non mi preoccuperei di ciò ora. Inoltre ora che vedo il metodo non dovrebbe essere static.